hw/xbox/xbox.c

```diff
@@ -227,11 +227,62 @@ static void xbox_memory_init(PCMachineState *pcms,
     xbox_flash_init(rom_memory);
 }
 
+uint8_t* load_eeprom(void)
+{
+    char *filename;
+    int fd;
+    int rc;
+    int eeprom_file_size;
+    const int eeprom_size = 256;
+
+    uint8_t *eeprom_data = g_malloc(eeprom_size);
+
+    const char *eeprom_file = object_property_get_str(qdev_get_machine(),
+                                                      "eeprom", NULL);
+    if ((eeprom_file != NULL) && *eeprom_file) {
+        filename = qemu_find_file(QEMU_FILE_TYPE_BIOS, eeprom_file);
+        assert(filename);
+
+        eeprom_file_size = get_image_size(filename);
+        if (eeprom_size != eeprom_file_size) {
+            fprintf(stderr,
+                    "qemu: EEPROM file size != %d bytes. (Is %d bytes)\n",
+                    eeprom_size, eeprom_file_size);
+            g_free(filename);
+            exit(1);
+            return NULL;
+        }
+
+        fd = open(filename, O_RDONLY);
```

Comment:
nit: might still want the binary flag to not break on shitty platforms (Windows in particular). not sure which `open` variant is exposed here and if QEMU deals with this.